src/js/spatial-navigation.js

```diff
@@ -0,0 +1,496 @@
+class SpatialNavigation {
+  /**
+   * Constructs a SpatialNavigation instance with initial settings.
+   * Initializes key codes for navigation, sets up the player instance,
+   * and prepares the spatial navigation system.
+   *
+   * @class
+   * @param {Object} player - The Video.js player instance to which the spatial navigation is attached.
+   * @param {Component|null} initialFocusedComponent - The component that should initially have focus
+   *                                                   when the spatial navigation system starts.
+   *                                                   If null or not provided, no component will be initially focused.
+   */
+  constructor(player, initialFocusedComponent) {
+    this.KEY_CODE = { 37: 'left', 38: 'up', 39: 'right', 40: 'down', 415: 'play', 19: 'pause', 417: 'ff', 412: 'rw' };
+    this.player = player;
+    this.components = new Set();
+    this.focusableComponents = [];
+    this.isListening = false;
+    this.isPaused = false;
+    this.eventListeners = [];
+    this.onKeyDown = this.onKeyDown.bind(this);
+    this.currentFocus = initialFocusedComponent || null;
+    this.lastFocusedComponent = null;
+    // The number of seconds the `step*` functions move the timeline.
+    this.STEP_SECONDS = 5;
+  }
+
+  /**
+   * Starts the spatial navigation by adding a keydown event listener to the video container.
+   * This method ensures that the event listener is added only once.
+   */
+  start() {
```

Comment:
This function is never called, preventing spatial navigation from working.

Reply:
Thank you for pointing this out, @amtins. The decision to not automatically start the spatial navigation was intentional, to provide developers with the flexibility to enable it based on their specific application requirements. There are scenarios, such as when using a spatial-navigation-polyfill, where it might be beneficial to have the "videojs spatialNavigation" functionality integrated and available but not active.

For more information on how to implement and configure spatial navigation, including how to manually start it, please refer to our draft documentation here: [Spatial Navigation Guide](https://github.com/bzizmo/videojs.com/blob/main/src/mdx-pages/guides/spatial-navigation.mdx).